design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,330 @@
+- **Errors**:
+  1. *403- Forbidden*: The requesting user is forbidden from accessing the configurations or functionality (like generating credentials).
+  2. *400- Bad request*: The request is missing information necessary for the template’s expansion.
+  3. *500- Internal server error*: The lakeFS server cannot access some provided template locations.
+
+---
+
+## Possible Flows
+
+### Example template
+
+**Databricks Spark configurations**  
+Name: *databricksConfig.props.tt*
+```properties
+spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem
+{{with $creds := new_credentials}}
+spark.hadoop.fs.s3a.access_key={{$creds.Key}}
+spark.hadoop.fs.s3a.secret_key={{$creds.Secret}}
+{{end}}
+spark.hadoop.fs.s3a.endpoint={{ .querystring.lakefs_url }}
+spark.hadoop.fs.s3a.path.style.access=true
+```
+
+**Local Metastore configurations**
+Name: *localMetastoreConfig.xml.tt*
+```xml
+<configuration>
+    <property>
+        <name>fs.s3a.path.style.access</name>
+        <value>true</value>
+    </property>
+    <property>
+        <name>fs.s3.impl</name>
+        <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
+    </property>
+    <property>
+        <name>fs.s3a.endpoint</name>
+        <value>{{ .querystring.lakefs_url }}</value>
+    </property>
+    {{with $creds := new_credentials}}
+    <property>
+        <name>fs.s3a.access.key</name>
+        <value>{{$creds.Key}}</value>
+    </property>
+    <property>
+        <name>fs.s3a.secret.key</name>
+        <value>{{$creds.Secret}}</value>
+    </property>
+    {{end}}
+</configuration>
+```
+
+### Happy flow - All steps
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to type the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user specifies the location of the bucket (after they validated that the lakeFS role has the right permissions and that the bucket has the correct policy), and clicks ‘OK’.
+   * An object counter will show the progress of the import process and will signal once it’s over.
+5. The wizard asks the user for their lakeFS endpoint (and will show a default placeholder pointing to the current URL).
+6. The wizard will send a GET request to the templating service with a query string of the format:  
+    ```
+    ?lakefs_url=https://my-lakefs.io&template_location=databricksConfig.props.tt
+    ```
+7. The templating service will fetch the templates from the provided locations, expand the `.querystring.lakefs_url` parameter for both templates and return a response of the format:
+    ```properties
+    spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem
+    spark.hadoop.fs.s3a.access_key=ACCESSKEYDONTTELL
+    spark.hadoop.fs.s3a.secret_key=SECRETKEYDONTTELL
```

Comment:
This kinds of break our promise when we create credentials: `This is the only time where you can see your credentials - make sure to copy them......`. I don't think that we shouldn't do it, only that we should probably rephrase that sentence.

Reply:
Good point.
Will add that